backend/internal/system/database/provider/dbprovider.go

```diff
@@ -108,3 +163,28 @@ func getDBConfig(dataSource config.DataSource) dbConfig {
 
 	return dbConfig
 }
+
+// Close gracefully closes all database connections.
+func (d *DBProvider) Close() error {
+	var errs []error
+
+	if d.identityClient != nil {
+		if err := d.identityClient.Close(); err != nil {
+			errs = append(errs, fmt.Errorf("failed to close identity client: %w", err))
```

Comment:
We can use `errors.Join` to concat multiple errors.  Shall we use that instead of current approach?

Also saw some other place where you are string concatinating two errors. Shall we fix that as well?